### PR TITLE
Adding documentation website

### PR DESCRIPTION
**Pull request by naskio**
## Summary 
- Adding website at `docs/`.
- Updating `README.md`.
- Moving `example/` folder and other markdown files into `docs/content/`.
- Adding GitHub workflow to auto-deploy the website into Pages `.github/workflows/deploy-website.yml`.
## ToDo
- [x] Create email address: `fasttrackml@gr-oss.io`
- [x] Review this PR
- [ ] Enable GitHub Pages + set domain to https://fasttrackml.io/
- [ ] Merge this PR
- [x] Configure domain name [fasttrackml.io](http://fasttrackml.io/)

You can preview the website at [https://naskio.github.io/fasttrackml/](https://naskio.github.io/fasttrackml/)

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/FML-6) by [Unito](https://www.unito.io)

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+    title: 'FastTrackML',
+    tagline: 'Experiment tracking server focused on speed and scalability',
+    favicon: 'favicon.ico',
+    onBrokenLinks: 'throw',
+    onBrokenMarkdownLinks: 'warn',
+
+    url: 'https://naskio.github.io',
```
Comment (jgiannuzzi) [inline docs/docusaurus.config.js:15]: could you please update the URL?

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    projectName: 'fasttrackml',
+
+    i18n: {
+        defaultLocale: 'en',
+        locales: ['en'],
+    },
+
+    customFields: {
+        email: 'fasttrackml@gr-oss.io',
+        newIssueUrl: 'https://github.com/naskio/fasttrackml/issues/new',
```
Comment (jgiannuzzi) [inline docs/docusaurus.config.js:28]: could you please point this to our repo?

File: .github/workflows/deploy-website.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            echo "command=install" >> $GITHUB_OUTPUT
+            exit 0
+          elif [ -f "package.json" ]; then
+            echo "manager=npm" >> $GITHUB_OUTPUT
+            echo "command=ci" >> $GITHUB_OUTPUT
+            exit 0
+          else
+            echo "Unable to determine packager manager"
+            exit 1
+          fi
```
Comment (jgiannuzzi) [inline .github/workflows/deploy-website.yml:45]: is there any specific benefit in this dynamic detection? could we maybe just assume it's going to be yarn?

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                        ],
+                    },
+                ],
+                logo: {
+                    alt: 'G-Research Open-Source Software',
+                    src: 'logo/organization.png',
+                    srcDark: 'logo/organization-dark.png',
+                    href: 'https://opensource.gresearch.co.uk/',
+                },
+                copyright: `Copyright © ${new Date().getFullYear()} FastTrackML`,
```
Comment (jgiannuzzi) [inline docs/docusaurus.config.js:128]: the copyright should be attributed to G-Research

**Review by jgiannuzzi — COMMENTED** [review]
Awesome stuff, thank you @naskio!

I have left a couple of comments below.

I'm also wondering whether we should maybe put the whole website into `docs/website` rather than putting it in `docs` and having to move the existing docs into `docs/content`?

**Comment by naskio** [comment]
Hi @jgiannuzzi,
Thank you for your feedback. I have made the necessary changes, including using the new identity (logo, colors, ...).
The only remaining point is the project structure.
In my opinion, it makes sense to have the website in the `docs/` folder and all content-related files (markdown, …) in `docs/content/`. However, we can use a different structure if needed.

File: docs/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+    title: 'FastTrackML',
+    tagline: 'Experiment tracking server focused on speed and scalability',
+    favicon: 'favicon.ico',
+    onBrokenLinks: 'throw',
+    onBrokenMarkdownLinks: 'warn',
+
+    url: 'https://G-Research.github.io',
+    baseUrl: '/fasttrackml/',
```
Comment (jgiannuzzi) [inline docs/docusaurus.config.js:16]: I think the URL will be https://fasttrackml.io/
@c-rindi could you please confirm?

Comment (naskio) [reply docs/docusaurus.config.js:16]: Yes, I guess that is the one.

**Comment by jgiannuzzi** [comment]
what about copying the content during the CI workflow?

**Comment by naskio** [comment]
Fortunately, it's possible to configure Docusaurus to do that, we don't need to customize the CI.
We have now the following structure:
```
...
website/
docs/
...
```
We can later generate the website based on `docs/` content without any additional headaches.

File: website/src/components/Index/QuickStartSection/images/main.png
Comment (jgiannuzzi) [inline website/src/components/Index/QuickStartSection/images/main.png:1]: Could you please link this to the existing image from the quickstart, or at worst copy it over?

File: website/src/components/Index/QuickStartSection/images/runs.png
Comment (jgiannuzzi) [inline website/src/components/Index/QuickStartSection/images/runs.png:1]: Same comment for this image

File: website/src/components/Index/QuickStartSection/images/main.png
Comment (naskio) [reply website/src/components/Index/QuickStartSection/images/main.png:1]: I generated new images to match the example output. There is a slight difference between the old images and the new ones.

**Review by jgiannuzzi — APPROVED** [review]

**Review by suprjinx — APPROVED** [review]

**Pull request merged**